dcmgr/lib/dcmgr/endpoints/12.03/instances.rb

```diff
@@ -186,8 +186,15 @@ def set_monitoring_parameters(instance)
 
   get '/:instance_id/password' do
     @instance = find_by_uuid(:Instance, params[:instance_id])
-    respond_with(R::InstancePassword.new(@instance).generate)
-  end
+    theresponse = R::InstancePassword.new(@instance).generate
+
+    if Dcmgr::Configurations.dcmgr.windows.delete_password_on_request
+      logger.info "Deleting encrypted_password for #{:instance_id}"
+      @instance.set({ :encrypted_password => nil }).save_changes
```

Comment:
Something like this? https://github.com/axsh/wakame-vdc/commit/c589c0cd188240188333253311b46812e4e864d0

(I'm about to test it now...) 
@unakatsuo